dotenv/dotenv_test.go

```diff
@@ -36,3 +38,76 @@ func TestNewBackend(t *testing.T) {
 		assert.Equal(t, "my value", string(result))
 	})
 }
+
+func TestNewBackend_Multiple(t *testing.T) {
+	b := dotenv.NewBackend("testdata/.env", "testdata/.env.other")
+
+	t.Run("Key format unexpected", func(t *testing.T) {
+		result, err := b.Get(context.TODO(), "invalid key with space")
+		assert.EqualError(t, err, "dotenv variable format expected \\w+, \"invalid key with space\" given")
+		assert.Equal(t, "", string(result))
+	})
+
+	t.Run("Key Not Found", func(t *testing.T) {
+		result, err := b.Get(context.TODO(), "test")
+		assert.EqualError(t, err, "key not found")
+		assert.Equal(t, "", string(result))
+	})
+
+	t.Run("Key Matched", func(t *testing.T) {
+		result, err := b.Get(context.TODO(), "my_string")
+		assert.NoError(t, err)
+		assert.Equal(t, "my other value", string(result))
+	})
+}
+
+func TestGetBackend(t *testing.T) {
+	t.Run("One Backend", func(t *testing.T) {
+		backends := dotenv.GetBackends("testdata/.env")
+		assert.Len(t, backends, 1)
+	})
+
+	t.Run("Two Backend", func(t *testing.T) {
+		backends := dotenv.GetBackends("testdata/.env", "testdata/.env.other")
+		assert.Len(t, backends, 2)
+	})
+
+	t.Run("File Not found", func(t *testing.T) {
+		backends := dotenv.GetBackends("testdata/.env", "unexisting.file")
+		assert.Len(t, backends, 1)
+	})
+}
+
+// This function will replace os.Args in order to simulate only wanted flags
+func replaceOSArgs(flags ...string) func() {
+	originalArgs := os.Args
+	os.Args = append([]string{originalArgs[0]}, flags...)
+	return func() {
+		os.Args = originalArgs
+	}
+}
+
+func TestGetBackendsFromFlag(t *testing.T) {
```

Comment:
i think there should be a test on the order because the order is important for the overloading

Reply:
the https://github.com/etf1/go-config/pull/5/files#diff-fe3917089828805f278ade48332ccb8dR102 will implicitly test override order